File: old_testenv/new_testenv/test-lib-testenv-sh/utils/bin/delete_collections.py
#!/usr/bin/python

import re
import sys
import vapi_interface
import velocityAPI


def all_collections():
   """Return a list of all collection names."""
   return velocityAPI.VelocityAPI().repository_list_xml().xpath(
      'vse-collection/@name')


def delete_collections_matching(wildcard):
   """Delete every existing collection matching a wildcard."""
   yy = vapi_interface.VAPIINTERFACE()
   for collection_name in filter(re.compile(wildcard).match, all_collections()):
      print 'deleting', collection_name, '...',
      sys.stdout.flush()
      if yy.cleanup_collection(collection_name, delete=True):
         print 'done.'
      sys.stdout.flush()


if __name__ == "__main__":
   if len(sys.argv) > 1:
      for wildcard in sys.argv[1:]:
         delete_collections_matching(wildcard)
   else:
      print 'usage:', sys.argv[0], 'wildcard [wildcard ...]'

File: old_testenv/new_testenv/test-lib-testenv-sh/utils/bin/delete-collections.rb
#!/usr/bin/env ruby

$LOAD_PATH << ENV['TEST_ROOT']+'/lib'

require 'testenv'
require 'collection'
require 'pmap'

def all_collections
  vapi = Vapi.new(TESTENV.velocity, TESTENV.user, TESTENV.password)
  vapi.repository_list_xml.xpath('/vce/vse-collection').map{|e| e['name'].to_s}
end

def delete_collections_matching(wildcard)
  matching_collections = all_collections.select {|a| a =~ /#{wildcard}/}
  if matching_collections.empty?
    puts "No collections found matching #{wildcard}"
  else
    puts "Deleting #{matching_collections.length} collections matching #{wildcard} ..."
    matching_collections.peach {|name| Collection.new(name).delete}
  end
end

if __FILE__ == $0
  if ARGV.length == 0
    puts 'usage: delete-collections.rb wildcard [wildcard ...]'
  else
    ARGV.each {|wildcard| delete_collections_matching(wildcard)}
  end
end
